pyqtgraph/opengl/GLGraphicsItem.py

```diff
@@ -31,14 +31,14 @@
 class GLGraphicsItem(QtCore.QObject):
     _nextId = 0
     
-    def __init__(self, parentItem=None):
+    def __init__(self, parentItem: 'GLGraphicsItem' = None):
         super().__init__()
         self._id = GLGraphicsItem._nextId
         GLGraphicsItem._nextId += 1
         
-        self.__parent = None
+        self.__parent: GLGraphicsItem | None = None
```

Comment:
Love to see type annotations, but I'm not sure we can use this syntax, as we support Python 3.8 still.  We will likely add typing-extensions as a dependency at some point but we are not there yet.

Reply:
To clarify, python3.8 as a runtime happily ignores typehints during execution. It is confirmed by the tests. Typehints needed for tooling you are using. Are you telling that maintaining support for python 3.8 means not only runtime but also all tools that have no clue about anything above 3.8?

Reply:
oh nice, I got that in my head that it would raise syntax errors or something; happy to restrict type checking to the current/newest version of python.